app/src/main/java/com/example/criminalintentapp/CrimeFragment.kt

```diff
@@ -97,13 +111,21 @@ class CrimeFragment : Fragment() {
 
     private fun updateUI() {
         titleField.setText(crime.title)
-        dateButton.text = crime.date.toString()
         solvedCheckBox.apply {
             isChecked = crime.isSolved
             jumpDrawablesToCurrentState()
         }
     }
 
+    override fun onFragmentResult(requestCode: String, result: Bundle) {
+        when (requestCode) {
+            DIALOG_DATE -> {
+                Log.d(TAG, "RECEIVED RESULT FOR $requestCode")
+                crime.date = DatePickerFragment.getSelectedDate(result)
```

Comment:
This looks a bit weird. Is this the way the book recommends getting the selected date?

Reply:
No, this is a solution that I found on the internet for passing the date between fragments using setFragmentResultlListener(), setFragmentResult() and onFragmentResult(). The book used callbacks interface and deprecated function setTargetFragment